meldingen_core/token.py

```diff
@@ -42,3 +45,23 @@ async def __call__(self, melding_id: int, token: str) -> T:
             raise TokenExpiredException()
 
         return melding
+
+
+class BaseTokenInvalidator(Generic[T], metaclass=ABCMeta):
+    _repository: BaseMeldingRepository[T]
+
+    def __init__(self, repository: BaseMeldingRepository[T]):
+        self._repository = repository
+
+    async def __call__(self, melding: T) -> T:
+        if not melding.state in self.allowed_states:
+            raise InvalidStateException()
+
+        melding.token = None
+        await self._repository.save(melding)
+
+        return melding
+
+    @property
+    def allowed_states(self) -> list[str]:
```

Comment:
Wil je dat deze wordt ge-override in de class die hiervan erft?
Indien ja, zou je deze method dan ook nog abstract kunnen maken?